src/core/providers/ogr/qgsogrfeatureiterator.cpp

```diff
@@ -85,7 +95,7 @@ QgsOgrFeatureIterator::QgsOgrFeatureIterator( QgsOgrFeatureSource *source, bool
       return;
     }
 
-    if ( !mSource->mSubsetString.isEmpty() )
+    if ( ! mDoNotResetReading && !mSource->mSubsetString.isEmpty() )
```

Comment:
to avoid the double negation, if we changed the variable to be mResetReadingAllowed that could make things slightly more obvious